encodings/dict/src/builders/primitive.rs

```diff
@@ -64,7 +64,7 @@ where
         Self {
             lookup: HashMap::with_hasher(FxBuildHasher),
             values: BufferMut::<T>::empty(),
-            values_nulls: NullBufferBuilder::new(0),
+            values_nulls: BitBufferMut::empty(),
```

Comment:
TODO: use MaskMut from https://github.com/vortex-data/vortex/pull/4954 will have the same lazy allocation logic